src/Symfony/Component/HttpClient/Tests/CurlHttpClientTest.php

```diff
@@ -42,4 +43,49 @@ public function testTimeoutIsNotAFatalError()
 
         parent::testTimeoutIsNotAFatalError();
     }
+
+    public function testOverridingRefererUsingCurlOptions()
+    {
+        $httpClient = $this->getHttpClient(__FUNCTION__);
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Could not override "CURLOPT_REFERER" param, please use "headers" option.');
+
+        $httpClient->request('GET', 'http://localhost:8057/', [
+            'extra' => [
+                'curl' => [
+                    \CURLOPT_REFERER => 'Banana',
+                ],
+            ],
+        ]);
+    }
+
+    public function testOverridingHttpMethodUsingCurlOptions()
+    {
+        $httpClient = $this->getHttpClient(__FUNCTION__);
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Could not override method using extra.curl options.');
+
+        $httpClient->request('POST', 'http://localhost:8057/', [
+            'extra' => [
+                'curl' => [
+                    \CURLOPT_HTTPGET => true,
+                ],
+            ],
+        ]);
+    }
+
+    public function testDefiningAdditionalCurlOptions()
+    {
+        $httpClient = $this->getHttpClient(__FUNCTION__);
+
+        $response = $httpClient->request('POST', 'http://localhost:8058/', [
+            'extra' => [
+                'curl' => [
+                    \CURLOPT_PORT => 8057,
+                ],
+            ],
+        ]);
+
+        $this->assertEquals(200, $response->getStatusCode());
```

Comment:
assertSame - but the test case is not legit also: the port should be given in the URL, not via an option, isn't it?

Reply:
fixed.